pkg/configs/db/postgres/postgres.go

```diff
@@ -214,17 +240,65 @@ func (d DB) findRulesConfigs(filter squirrel.Sqlizer) (map[string]configs.Versio
 
 // GetAllRulesConfigs gets all alertmanager configs for all users.
 func (d DB) GetAllRulesConfigs() (map[string]configs.VersionedRulesConfig, error) {
-	return d.findRulesConfigs(activeConfig)
+	return d.findRulesConfigs(allConfigs)
 }
 
 // GetRulesConfigs gets all the alertmanager configs that have changed since a given config.
 func (d DB) GetRulesConfigs(since configs.ID) (map[string]configs.VersionedRulesConfig, error) {
 	return d.findRulesConfigs(squirrel.And{
-		activeConfig,
+		allConfigs,
 		squirrel.Gt{"id": since},
 	})
 }
 
+// GetLastConfig gets a last configuration (active or deleted).
+func (d DB) GetLastConfig(userID string) (configs.View, error) {
```

Comment:
I have a feeling this function isn't necessary: `GetConfig` is good enough for the purpose.